# ãƒ—ãƒ­ã‚°ãƒ©ãƒ ã‚ºãƒ»ãƒãƒ–ãƒ»ãƒãƒ¼ã‚ºï¼ˆPrograms Hub Nodesï¼›ãƒ—ãƒ­ã‚°ãƒ©ãƒ ä¸­å¿ƒåœ°ç¯€ï¼‰
extends Node


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ãƒ¡ãƒ¢ãƒªé–¢é€£
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼

# å…ˆç¥–ã®è¾æ›¸ã‚­ãƒ£ãƒƒã‚·ãƒ¥
var ancestors = {}


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ãƒãƒ¼ãƒ‰ãƒ»ãƒ‘ã‚¹é–¢é€£
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


# ãƒ—ãƒ­ã‚°ãƒ©ãƒãƒ¼ãƒ»ãƒãƒ–å–å¾—
func hub():
	return MonkeyHelper.find_ancestor_child(
			self,
			"ğŸŒProgrammer/ğŸµMonkey",
			self.ancestors)


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# æº–å‚™
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


func _ready():
	self.hub().bg_musics = MonkeyHand.create(
			self.hub().get_director_hub().musician_bg_musics())		# æ¢ã™å ´æ‰€

	self.hub().images = MonkeyHand.create(
			self.hub().get_director_hub().illustrator())			# æ¢ã™å ´æ‰€

	self.hub().sound_fx = MonkeyHand.create(
			self.hub().get_director_hub().musician_sound_fx())			# æ¢ã™å ´æ‰€

	self.hub().telops = MonkeyHand.create(
			self.hub().get_director_hub().telop_coordinator())	# æ¢ã™å ´æ‰€
